### DISCUSSION TITLE
Using 'next/dynamic' outside of next.js

### DISCUSSION THREAD

**Discussion by jrmyio** [vercel/next.js#26989] [Help]
When building a component library I am rendering certain components in different environments. For dynamic imports I am utilizing `@loadable/component` but this doesn't render in React.js SSG, so I am considering to use `next/dynamic`.

However, using  'next/dynamic' outside of a next.js application seems to return an `object` with a `render` property (which is a function) and some other properties, rather than a React Component.

How can you make 'next/dynamic' work in a different environment (like `@loadable/component`) and what part of next.js makes it return an actual react component rather than the `object`.

Currently my work around is to change next.config.js webpack configuration by adding:
`config.resolve.alias['@loadable/component'] = 'next/dynamic';`

Since both main components have pretty much the same API this seems to work.

**Comment by jamesmosier** [comment]
`next/dynamic` is a fork of https://github.com/jamiebuilds/react-loadable. You might be able to use that instead.

Or alternatively `@loadable/server` might work: https://loadable-components.com/docs/server-side-rendering/

`next/dynamic` is only meant to be used with Next.js applications.
